### DISCUSSION TITLE
Regarding text.html.php

### DISCUSSION THREAD

**Discussion by PolVilniusTech** [github-linguist/linguist#5315] [Classification] [answered]
# Some Scripting language
codemirror_mime_type: application/x-httpd-php

[Mime type request](https://github.com/PolVilniusTech/coding.php/blob/main/Mime.png)
fig. mime-type of the *.php file

# Full-featured Programming language
codemirror_mime_type: application/x-php

or vica-versa of course.

**Comment by lildude** [comment]
🤔 I'm not 100% sure what you're asking for with this request. It looks like you want to change the mime-type for PHP files.

If this is correct, then this is completely outside of the control of Linguist.  The `codemirror_mime_type` entries as defined in the `languages.yml` file come from https://github.com/codemirror/CodeMirror/blob/master/mode/meta.js#L105 which is _not_ a Linguist project.

All other uses of mime-types within Linguist come from the mini_mime gem: https://github.com/discourse/mini_mime

If this is _not_ what you are asking for, can you please provide more details about what improvement you're asking for. Please be as clear and thorough as possible.

**Comment by PolVilniusTech** [comment]
Well my problem were only that there is no programming language sign in the github.com repo, when there exists some directories with files of this programming language.

Probably I understood the [languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) in Your project not in right way.

nvm:)

**Comment by lildude** [comment]
Oh ok. That makes a bit more sense. If the repo is public, provide a link and we can have a look and see what's going on.

As this is more of a topic for discussions, I'm going to move this over to the Classification section of the discussions.

**Reply by lildude** [reply, marked as answer]
I think I know which repo you're seeing this on: https://github.com/PolVilniusTech/coding.php

The reason you're not seeing any language information is because all your code is in a `samples` directory. Linguist considers the content of this directory as documentation:

https://github.com/github/linguist/blob/79b3da822eac18d334e012e97442a654955a08a3/lib/linguist/documentation.yml#L32-L33

So you will need to implement a [documentation override](https://github.com/github/linguist/blob/master/docs/overrides.md#documentation) to tell Linguist to _not_ consider the directory as documentation.

**Reply by PolVilniusTech** [reply]
Aha.
Thank You.
